model/system_metrics.go

```diff
@@ -258,6 +275,42 @@ func (sm *SystemMetrics) appendSystemMetricsChunkKey(ctx context.Context, metric
 	return errors.Wrapf(err, "problem appending system metrics data chunk to %s", sm.ID)
 }
 
+// Download returns an io.Reader for the system metrics data of the specified type.
```

Comment:
This should say SystemMetricsReader rather than io.Reader because even though it does implement the io.Reader interface, it also implements other methods.